src/crypto/ecdh.rs

```diff
@@ -19,87 +19,196 @@ const ANON_SENDER: [u8; 20] = [
     0x20, 0x20, 0x20, 0x20,
 ];
 
-const SECRET_KEY_LENGTH: usize = 32;
-
-/// Secret key for ECDH with Curve25519, the only combination we currently support.
+/// Secret key for ECDH
 #[derive(Clone, PartialEq, Eq, Zeroize, ZeroizeOnDrop)]
-pub struct SecretKey {
-    /// The secret point.
-    pub secret: [u8; 32],
-    pub hash: HashAlgorithm,
-    pub oid: Vec<u8>,
-    pub alg_sym: SymmetricKeyAlgorithm,
+pub enum SecretKey {
+    // ECDH with Curve25519
+    Curve25519Legacy {
```

Comment:
why legacy?

Reply:
I was going with the naming in: https://www.ietf.org/archive/id/draft-ietf-openpgp-crypto-refresh-13.html#section-5.5.5.6.1.1

However, I'm unsure if the two Curve 25519 formats in that draft even need to be modeled (and named) as separate `SecretKey` variants. Maybe they don't, then this name would be wrong.

Reply:
I think we should keep calling this `Curve25519` and only rename it to legacy when introducing v6 keys and the non legacy format

Reply:
Agreed